jsonnet/kube-prometheus/components/external-mixins.libsonnet

```diff
@@ -0,0 +1,58 @@
+local defaults = {
+  namespace: error 'must provide namespace',
+  commonLabels:: {
+    'app.kubernetes.io/name': 'kube-prometheus',
+    'app.kubernetes.io/component': 'external-mixins',
+    'app.kubernetes.io/part-of': 'kube-prometheus',
+  },
+  ruleLabels: {
+    role: 'alert-rules',
+  },
+};
+
+function(params) {
+  local m = self,
+  config:: defaults + params,
+
+  local prometheusRule(item) = {
+    apiVersion: 'monitoring.coreos.com/v1',
+    kind: 'PrometheusRule',
+    metadata: {
+      labels: m.config.commonLabels + m.config.ruleLabels,
+      name: item.name + '-rules',
+      namespace: if std.objectHas(item, 'namespace') then item.namespace else m.config.namespace,
+    },
+    spec: {
+      local r = if std.objectHasAll(item.mixin, 'prometheusRules') then item.mixin.prometheusRules.groups else [],
+      local a = if std.objectHasAll(item.mixin, 'prometheusAlerts') then item.mixin.prometheusAlerts.groups else [],
+      groups: a + r,
+    },
+  },
```

Comment:
Why complicate it more than it is in https://github.com/prometheus-operator/kube-prometheus/blob/main/examples/prometheus-additional-alert-rule-example.jsonnet ?